rune/libenclave/internal/runtime/pal/pal_linux.go

```diff
@@ -3,28 +3,66 @@ package enclave_runtime_pal // import "github.com/inclavare-containers/rune/libe
 import (
 	"encoding/binary"
 	"fmt"
+	"strings"
 	"github.com/go-restruct/restruct"
 	"github.com/inclavare-containers/rune/libenclave/attestation"
 	"github.com/inclavare-containers/rune/libenclave/attestation/sgx"
 	_ "github.com/inclavare-containers/rune/libenclave/attestation/sgx/ias"
 	"github.com/inclavare-containers/rune/libenclave/intelsgx"
+	"github.com/alibaba/inclavare-containers/epm/pkg/epm-api/v1alpha1"
+	"github.com/inclavare-containers/rune/libenclave/epm"
+
 	"log"
 	"os"
 )
 
 const (
 	palApiVersion = 3
+	InvalidEpmID string = "InvalidEPM"
 )
 
 func (pal *enclaveRuntimePal) Init(args string, logLevel string) error {
-	api := &enclaveRuntimePalApiV1{}
+	var enclaveinfo *v1alpha1.Enclave
+	var err error
+	var addr int64 = 0
+	var fd int = -1
+
+	pal.ID = InvalidEpmID
+
+	api := &enclaveRuntimePalApiV3{}
 	ver := api.get_version()
 	if ver > palApiVersion {
 		return fmt.Errorf("unsupported pal api version %d", ver)
 	}
 	pal.version = ver
 
-	return api.init(args, logLevel)
+	/* FIXME: If EPM provides epm existence detect API, the static check will be substituted. Enclave pool
```

Comment:
Plz restructure this Init() function as following:

```
func (pal *enclaveRuntimePal) Init(args string, logLevel string) error {
        api := &enclaveRuntimePalApiV1{}
	ver := api.get_version()
	if ver > palApiVersion {
		return fmt.Errorf("unsupported pal api version %d", ver)
	}
	pal.version = ver

        err := nil
        if pal.version >= 3 {
		err = createEnclaveShareMapping(args)
	}

        return err
```

In other words, move all epm staffs into createEnclaveShareMapping().